### PR TITLE
exchanges: upgrade bittrex api

### PR DESCRIPTION
**Pull request by buck54321**
v1.1 -> v3

Part of #1810

Still running long tests. I saw a couple of sequence errors over the course of a few hours, so I wanna watch for a while longer.

### CODE REVIEW

File: exchanges/exchanges.go [hunk 1]
```diff
@@ -1616,38 +1639,37 @@ func (bittrex *BittrexExchange) msgHandler(msg signalr.Message) {
...
 	bittrex.queue = make([]*BittrexOrderbookUpdate, 0)
 	bittrex.orderSeq = 0
 	bittrex.orderMtx.Unlock()
+
+	err := bittrex.connectSignalr(&signalrConfig{
+		host:       "socket-v3.bittrex.com",
+		protocol:   "1.5",
+		endpoint:   "/signalr",
+		msgHandler: bittrex.msgHandler,
+	})
```
Comment (chappjc) [inline exchanges/exchanges.go:1652]: Thoughts about migrating the signalr repo to https://github.com/rainhq/signalr from the archived carterjones one? I was poking around and it seems active and even updated with [an example for bittrex v3](https://github.com/rainhq/signalr/blob/v2/examples/bittrex_v3/main.go).

Comment (buck54321) [reply exchanges/exchanges.go:1652]: Looks like that should work. Do you mind if I open an issue and follow up with the change? I wanna hit some other priorities first.

Comment (chappjc) [reply exchanges/exchanges.go:1651]: Yah, sounds good

**Comment by chappjc** [comment]
Shall I update the rate server with this as a test?

**Comment by buck54321** [comment]
Sure

**Comment by chappjc** [comment]
Running:

```
2021-04-27 15:50:10.498 [INF] SRVR: ExchangeBot monitoring coindesk, poloniex, dcrdex, binance, bittrex, dragonex, huobi, coinbase
2021-04-27 15:50:10.517 [TRC] XBOT: Initializing websocket connection for dcrdex
2021-04-27 15:50:10.739 [TRC] XBOT: index update received from coindesk with 3 indices, USD price for Bitcoin is 55087.498300
2021-04-27 15:50:10.740 [INF] SRVR: Update received from coindesk
2021-04-27 15:50:10.790 [TRC] XBOT: index update received from coinbase with 221 indices, USD price for Bitcoin is 55258.540000
2021-04-27 15:50:10.790 [INF] SRVR: Update received from coinbase
2021-04-27 15:50:10.875 [TRC] XBOT: Initializing websocket connection for poloniex
2021-04-27 15:50:10.990 [TRC] XBOT: Initializing websocket connection for bittrex
2021-04-27 15:50:11.041 [TRC] XBOT: exchange update received from dcrdex with a BTC price 0.003864, 
2021-04-27 15:50:11.042 [INF] SRVR: Update received from dcrdex
2021-04-27 15:50:11.080 [TRC] XBOT: Signalling candlestick update for binance, bin size 1h
2021-04-27 15:50:11.142 [TRC] XBOT: Signalling candlestick update for huobi, bin size 1h
2021-04-27 15:50:11.274 [TRC] XBOT: Signalling candlestick update for binance, bin size 1d
2021-04-27 15:50:11.355 [TRC] XBOT: Signalling candlestick update for huobi, bin size 1d
2021-04-27 15:50:11.368 [TRC] XBOT: Signalling candlestick update for poloniex, bin size 1d
2021-04-27 15:50:11.462 [TRC] XBOT: Signalling candlestick update for binance, bin size 1mo
2021-04-27 15:50:11.559 [TRC] XBOT: Signalling candlestick update for huobi, bin size 1mo
2021-04-27 15:50:11.571 [TRC] XBOT: exchange update received from poloniex with a BTC price 0.000000, 
2021-04-27 15:50:11.572 [INF] SRVR: Update received from poloniex
2021-04-27 15:50:11.633 [TRC] XBOT: exchange update received from binance with a BTC price 0.003854, 
2021-04-27 15:50:11.639 [INF] SRVR: Update received from binance. Current price: 212.64 USD
2021-04-27 15:50:11.654 [TRC] XBOT: Signalling candlestick update for poloniex, bin size 30m
2021-04-27 15:50:11.736 [TRC] XBOT: exchange update received from huobi with a BTC price 0.003859, 
2021-04-27 15:50:11.747 [INF] SRVR: Update received from huobi. Current price: 212.66 USD
2021-04-27 15:50:12.678 [TRC] XBOT: Signalling candlestick update for dragonex, bin size 1h
2021-04-27 15:50:13.052 [TRC] XBOT: Signalling candlestick update for dragonex, bin size 1d
2021-04-27 15:50:13.440 [TRC] XBOT: exchange update received from dragonex with a BTC price 0.000000, 
2021-04-27 15:50:13.452 [INF] SRVR: Update received from dragonex. Current price: 212.66 USD
2021-04-27 15:50:13.934 [INF] SRVR: Client has connected from 149.248.61.168:35006
2021-04-27 15:50:14.095 [INF] SRVR: Client has connected from 104.196.50.243:48936
2021-04-27 15:50:14.524 [INF] SRVR: Client has connected from 35.202.87.212:51474
2021-04-27 15:50:14.820 [INF] SRVR: Client has connected from 35.204.11.209:50120
2021-04-27 15:50:31.181 [TRC] XBOT: Signalling candlestick update for bittrex, bin size 1d
2021-04-27 15:50:31.181 [TRC] XBOT: exchange update received from bittrex with a BTC price 0.000000, 
2021-04-27 15:50:31.193 [INF] SRVR: Update received from bittrex. Current price: 212.66 USD
2021-04-27 15:50:31.373 [TRC] XBOT: Signalling candlestick update for bittrex, bin size 1h
2021-04-27 15:50:49.120 [TRC] XBOT: Signalling candlestick update for dragonex, bin size 1h
2021-04-27 15:50:49.505 [TRC] XBOT: exchange update received from dragonex with a BTC price 0.000000, 
2021-04-27 15:50:49.521 [INF] SRVR: Update received from dragonex. Current price: 212.66 USD
2021-04-27 15:51:25.761 [TRC] XBOT: last bittrex websocket update 0.232 seconds ago
2021-04-27 15:51:25.761 [TRC] XBOT: exchange update received from bittrex with a BTC price 0.003855, 
2021-04-27 15:51:25.783 [INF] SRVR: Update received from bittrex. Current price: 212.66 USD
2021-04-27 15:52:03.425 [TRC] XBOT: exchange update received from binance with a BTC price 0.003845, 
2021-04-27 15:52:03.447 [INF] SRVR: Update received from binance. Current price: 212.23 USD
2021-04-27 15:52:40.517 [TRC] XBOT: exchange update received from dcrdex with a BTC price 0.003864, 
2021-04-27 15:52:40.532 [INF] SRVR: Update received from dcrdex. Current price: 212.23 USD
```

I think the strange "BTC price 0.000000" did that before this though.

The bittrex candlestick updates also came about 20 seconds after the other exchanges.  maybe just bittrex?

**Comment by chappjc** [comment]
Running:

```
2021-04-27 15:50:10.498 [INF] SRVR: ExchangeBot monitoring coindesk, poloniex, dcrdex, binance, bittrex, dragonex, huobi, coinbase
2021-04-27 15:50:10.517 [TRC] XBOT: Initializing websocket connection for dcrdex
2021-04-27 15:50:10.739 [TRC] XBOT: index update received from coindesk with 3 indices, USD price for Bitcoin is 55087.498300
2021-04-27 15:50:10.740 [INF] SRVR: Update received from coindesk
2021-04-27 15:50:10.790 [TRC] XBOT: index update received from coinbase with 221 indices, USD price for Bitcoin is 55258.540000
2021-04-27 15:50:10.790 [INF] SRVR: Update received from coinbase
2021-04-27 15:50:10.875 [TRC] XBOT: Initializing websocket connection for poloniex
2021-04-27 15:50:10.990 [TRC] XBOT: Initializing websocket connection for bittrex
2021-04-27 15:50:11.041 [TRC] XBOT: exchange update received from dcrdex with a BTC price 0.003864, 
2021-04-27 15:50:11.042 [INF] SRVR: Update received from dcrdex
2021-04-27 15:50:11.080 [TRC] XBOT: Signalling candlestick update for binance, bin size 1h
2021-04-27 15:50:11.142 [TRC] XBOT: Signalling candlestick update for huobi, bin size 1h
2021-04-27 15:50:11.274 [TRC] XBOT: Signalling candlestick update for binance, bin size 1d
2021-04-27 15:50:11.355 [TRC] XBOT: Signalling candlestick update for huobi, bin size 1d
2021-04-27 15:50:11.368 [TRC] XBOT: Signalling candlestick update for poloniex, bin size 1d
2021-04-27 15:50:11.462 [TRC] XBOT: Signalling candlestick update for binance, bin size 1mo
2021-04-27 15:50:11.559 [TRC] XBOT: Signalling candlestick update for huobi, bin size 1mo
2021-04-27 15:50:11.571 [TRC] XBOT: exchange update received from poloniex with a BTC price 0.000000, 
2021-04-27 15:50:11.572 [INF] SRVR: Update received from poloniex
2021-04-27 15:50:11.633 [TRC] XBOT: exchange update received from binance with a BTC price 0.003854, 
2021-04-27 15:50:11.639 [INF] SRVR: Update received from binance. Current price: 212.64 USD
2021-04-27 15:50:11.654 [TRC] XBOT: Signalling candlestick update for poloniex, bin size 30m
2021-04-27 15:50:11.736 [TRC] XBOT: exchange update received from huobi with a BTC price 0.003859, 
2021-04-27 15:50:11.747 [INF] SRVR: Update received from huobi. Current price: 212.66 USD
2021-04-27 15:50:12.678 [TRC] XBOT: Signalling candlestick update for dragonex, bin size 1h
2021-04-27 15:50:13.052 [TRC] XBOT: Signalling candlestick update for dragonex, bin size 1d
2021-04-27 15:50:13.440 [TRC] XBOT: exchange update received from dragonex with a BTC price 0.000000, 
2021-04-27 15:50:13.452 [INF] SRVR: Update received from dragonex. Current price: 212.66 USD
2021-04-27 15:50:31.181 [TRC] XBOT: Signalling candlestick update for bittrex, bin size 1d
2021-04-27 15:50:31.181 [TRC] XBOT: exchange update received from bittrex with a BTC price 0.000000, 
2021-04-27 15:50:31.193 [INF] SRVR: Update received from bittrex. Current price: 212.66 USD
2021-04-27 15:50:31.373 [TRC] XBOT: Signalling candlestick update for bittrex, bin size 1h
2021-04-27 15:50:49.120 [TRC] XBOT: Signalling candlestick update for dragonex, bin size 1h
2021-04-27 15:50:49.505 [TRC] XBOT: exchange update received from dragonex with a BTC price 0.000000, 
2021-04-27 15:50:49.521 [INF] SRVR: Update received from dragonex. Current price: 212.66 USD
2021-04-27 15:51:25.761 [TRC] XBOT: last bittrex websocket update 0.232 seconds ago
2021-04-27 15:51:25.761 [TRC] XBOT: exchange update received from bittrex with a BTC price 0.003855, 
2021-04-27 15:51:25.783 [INF] SRVR: Update received from bittrex. Current price: 212.66 USD
2021-04-27 15:52:03.425 [TRC] XBOT: exchange update received from binance with a BTC price 0.003845, 
2021-04-27 15:52:03.447 [INF] SRVR: Update received from binance. Current price: 212.23 USD
2021-04-27 15:52:40.517 [TRC] XBOT: exchange update received from dcrdex with a BTC price 0.003864, 
2021-04-27 15:52:40.532 [INF] SRVR: Update received from dcrdex. Current price: 212.23 USD
```

I think the strange "BTC price 0.000000" did that before this though.

The bittrex candlestick updates also came about 20 seconds after the other exchanges.  maybe just bittrex?

**Review by chappjc — APPROVED** [review]

**Pull request merged**